### PR TITLE
Keep loaded bytes in parser

### PR DESCRIPTION
**Pull request by jkeiser**
This fixes issue #570 and also makes it so multiple `parser.load()` calls can reuse the same buffer. It doesn't offer much control over the buffer, but it fixes the present problem and is strictly better than before :)

### CODE REVIEW

File: tests/basictests.cpp [hunk 1]
```diff
@@ -53,514 +94,787 @@ bool number_test_small_integers() {
...
+      count++;
+    }
+    if (count != 2) { cerr << "parse_many returned " << count << " documents, expected 2" << endl; return false; }
+    return true;
   }
-  iter.move_to_value();
-  if(!iter.is_object()) {
-    printf("Value of image should be object\n");
-    return false;
+  // bool parser_parse_many_empty() {
```
Comment (lemire) [inline tests/basictests.cpp]: Is parser_parse_many_empty mean to be commented out?

Comment (jkeiser) [reply tests/basictests.cpp]: Yep. I wanted to add the test while I was in there if it was already succeeding, but it fails. It's not a new failure, though. Either there is a bug, or my expectations are wrong (I expect it to succeed, but not yield any documents).

File: tests/basictests.cpp [hunk 2]
```diff
@@ -892,8 +1079,7 @@ namespace dom_api {
     if (doc["a"].as_uint64_t().first != 1) { cerr << "Expected uint64_t(doc[\"a\"]) to be 1, was " << doc["a"].first << endl; return false; }
 
     UNUSED document::element val;
-    // tie(val, error) = doc["d"]; fails with "no viable overloaded '='" on Apple clang version 11.0.0
-    doc["d"].tie(val, error);
+    tie(val, error) = doc["d"];
```
Comment (lemire) [inline tests/basictests.cpp]: See the CI tests: this syntax fails under libc++...

Comment (lemire) [reply tests/basictests.cpp]: Annoyingly, I might add...

**Comment by lemire** [comment]
I like how the new libc++ tests are being tripped here.

Now, we can decide to forgo support for libc++... but I think this needs to be a deliberate choice.

File: tests/basictests.cpp (hunk 2 above)
Comment (jkeiser) [reply tests/basictests.cpp]: Yeah, I did some copy/pasting of a few tests when I moved it around, I'll restore what you have in there.

**Comment by jkeiser** [comment]
Yeah, I concur that we should support libc++. I'd rather not give up a whole platform (OS/X) unless we have to. Thanks for adding that :)

**Comment by lemire** [comment]
> I'd rather not give up a whole platform (OS/X) unless we have to.

[Also Android](https://developer.android.com/ndk/guides/cpp-support)...

"LLVM's libc++ is the C++ standard library that has been used by the Android OS since Lollipop, and as of NDK r18 is the only STL available in the NDK."

**Comment by lemire** [comment]
[And because you can build for Android using Visual Studio, this means that libc++ is used within Visual Studio](https://developercommunity.visualstudio.com/content/problem/207865/android-llvm-libc-static-library-stl-broken-with-v.html).

**Pull request merged**